pkg/driver/internal/inflight.go

```diff
@@ -45,26 +46,25 @@ func NewInFlight() *InFlight {
 
 // Insert inserts the entry to the current list of inflight requests.
 // Returns false when the key already exists.
-func (db *InFlight) Insert(entry Idempotent) bool {
+func (db *InFlight) Insert(volumeId string) bool {
 	db.mux.Lock()
 	defer db.mux.Unlock()
 
-	hash := entry.String()
-
-	_, ok := db.inFlight[hash]
+	_, ok := db.inFlight[volumeId]
 	if ok {
 		return false
 	}
 
-	db.inFlight[hash] = true
+	db.inFlight[volumeId] = true
 	return true
 }
 
 // Delete removes the entry from the inFlight entries map.
 // It doesn't return anything, and will do nothing if the specified key doesn't exist.
-func (db *InFlight) Delete(h Idempotent) {
+func (db *InFlight) Delete(volumeId string) {
 	db.mux.Lock()
 	defer db.mux.Unlock()
 
-	delete(db.inFlight, h.String())
+	delete(db.inFlight, volumeId)
```

Comment:
so `h.String()` is always equal to `volumeId`?

Reply:
No, previously the key in this cache is the hash of entire request( including volumeId and other fields like targetPath) so the driver has chance to process multiple requests for one volume at same time (Like same volumeId with different targetPath), which should not happen per [spec](https://github.com/container-storage-interface/spec/blob/master/spec.md#concurrency)
That's why I changed the key to volumeId.

Reply:
I guess my question is will the `delete` function work properly with the `volumeId` instead?

Reply:
Oh nvm I'm dumb heh.